src/main/java/com/iridium/iridiumskyblock/listeners/BucketListener.java

```diff
@@ -40,4 +48,44 @@ public void onBucketEvent(PlayerBucketEvent event) {
         player.sendMessage(StringUtils.color(IridiumSkyblock.getInstance().getMessages().cannotUseBuckets.replace("%prefix%", IridiumSkyblock.getInstance().getConfiguration().prefix)));
     }
 
+    @EventHandler
+    public void onClick(PlayerInteractEvent event) {
+        // I tested this on 1.17.1 so if it doesnt work on any version lower, they can suck it kekW
+
+        final Block clickedBlock = event.getClickedBlock();
+        if (clickedBlock == null)
+            return;
+
+        if (event.getHand() != EquipmentSlot.HAND)
+            return;
+
+        final ItemStack handItem = event.getItem();
+        if (handItem == null)
+            return;
+
+        if (handItem.getType() != Material.BUCKET)
+            return;
+
+        if (clickedBlock.getType() != Material.OBSIDIAN)
+            return;
+
+        final IslandManager manager = IridiumSkyblock.getInstance().getIslandManager();
+
+        final Optional<Island> islandOptional = manager.getIslandViaLocation(event.getClickedBlock().getLocation());
+        if (!islandOptional.isPresent())
+            return;
+
+        final User user = IridiumSkyblock.getInstance().getUserManager().getUser(event.getPlayer());
+
+        if (!manager.getIslandPermission(islandOptional.get(), user, PermissionType.BUCKET))
+            return;
+
+
+        event.getClickedBlock().setType(Material.AIR);
```

Comment:
it will work to all obsidians, u should check this obsidian from an lava

Reply:
so we don't player player break a nether portal with a bucket

Reply:
Perhaps make a database class that saves all the failed ones to a database :thinking:

Reply:
From what I know there's no spigot method to check if a block was placed or formed

Reply:
There is, Block form listener check if its new state is obsidian or not

Reply:
Then add to a database or hashmap or something

Reply:
Yeah but that's in the BlockFormListener event. There is no way to check a block's potential previous state outside of the event. The only alternative is to cache all the current formed blocks in the plugin. 

Saving all the formed blocks seems incredibly unnecessary and just caching it without saving it would reset every restart

Reply:
Its not going to save all formed blocks, only the ones that are obsidian, and putting it in a hashmap would reset it every restart but people would usually click it with a bucket within like 10 seconds to get it back so :shrug: